### DISCUSSION TITLE
Focus on cloudflare issues?

### DISCUSSION THREAD

**Discussion by acoreyj** [remix-run/remix#5127] [General]
Cloudflare and edge support is a huge feature and rightly called out a lot in blog posts and docs but currently there are a lot of gotchas that can come up when it comes to actually launching a site.

Some of these need to be discussed and/or documented but some already have fixes/features that don't seem to be getting any traction from the team.

The contributing guide says that "The Admins will review open bugfix PRs as part of Roadmap Planning" so it's a bit disheartening to see ones that you have been hoping for updates on sit for months. 

Some examples:
- https://github.com/remix-run/remix/pull/3330
- https://github.com/remix-run/remix/pull/4676
- https://github.com/remix-run/remix/issues/3590
- https://github.com/remix-run/remix/discussions/3066

**Comment by sergiodxa** [comment]
I agree, we need more love for Cloudflare support, it works but it doesn't as good as Node, for example SSR Streaming.

Templates for Node come with streaming but CF doesn't, not sure if this is a React issue but once `defer` is supported it will not be possible to use it with CF without enabling SSR Streaming.

This also means it's not possible to use React.lazy without ClientOnly.

**Reply by clgeoio** [reply]
You can use streaming with CF now, you just need to add this to your `wrangler.toml`
```
compatibility_date = "2021-09-29"
compatibility_flags = ["streams_enable_constructors"]
```

and change your `entry.server.tsx` to something like:
```
async handleRequest 
...

  const stream = await renderToReadableStream(
      <RemixServer context={remixContext} url={request.url} />,
  );

  return new Response(stream, {
    status: responseStatusCode,
    headers: responseHeaders,
  });

**Reply by kishanhitk** [reply]
@clgeoio I tried your solution but I am getting this error `TypeError: Invalid state: Controller is already closed`. Same issue as #5363

**Comment by alfl** [comment]
Related discussion: https://github.com/remix-run/remix/discussions/5226
